roles/katello_client/tasks/main.yml

```diff
@@ -1,16 +1,35 @@
 ---
+- name: 'Cleanup previous registrations'
+  include_tasks: cleanup.yml
+  when: katello_client_cleanup
+
 - name: 'Install bootstrap rpm'
   yum:
     name: "http://{{ katello_client_server }}/pub/katello-ca-consumer-latest.noarch.rpm"
     state: "present"
 
-- name: 'Register client with subscription-manager'
+- name: 'Register client with subscription-manager as user {{ katello_client_username }}'
   redhat_subscription:
     state: "present"
     org_id: "{{ katello_client_organization }}"
     environment: "{{ katello_client_environment }}"
     username: "{{ katello_client_username }}"
     password: "{{ katello_client_password }}"
+  when:
+    - katello_client_activationkey == ""
+
+- name: 'Register client with subscription-manager using AK {{ katello_client_activationkey }}'
+  redhat_subscription:
+    state: "present"
+    org_id: "{{ katello_client_organization }}"
+    activationkey: "{{ katello_client_activationkey }}"
+  when:
+    - katello_client_activationkey != ""
+
+- name: 'Enable extra repositories'
+  command: subscription-manager repos --enable {{ item }}
+  with_items:
+    "{{ katello_client_extra_repos }}"
```

Comment:
This line is a bit of a hanging chad here. Did you intend to have a `-` or shoudl it be on the line above it?

Reply:
either works, I think. on the line above is probably more readable.